Dockerfile

```diff
@@ -23,11 +23,19 @@ RUN cp -a /src/dnscrypt-proxy/example-* ./
 COPY dnscrypt-proxy.toml ./
 
 # ----------------------------------------------------------------------------
+FROM --platform=$BUILDPLATFORM cgr.dev/chainguard/go:1.20.2 as probe
 
+WORKDIR /src/dnsprobe
+
+ADD dnsprobe/ ./
+RUN CGO_ENABLED=0 go build -o /usr/local/bin/dnsprobe .
```

Comment:
You need to set `GOOS=$TARGETOS GOARCH=$TARGETARCH` here like we did above so golang compiles for the target platform.

Either that, or remove `--platform=$BUILDPLATFORM` from your `FROM` line so we pull the go image for our target platform and build with emulation. The former is probably faster to build.

Reply:
Good catch, let me fix that.

> build with emulation

Yeah, I tried that once and it's not an experience I'd like to repeat.